pkg/graph/parser/parser.go

```diff
@@ -75,18 +80,38 @@ func validateSchema(schema *spec.Schema, path string) error {
 	return nil
 }
 
-func getExpectedType(schema *spec.Schema) string {
+func getExpectedType(schema *spec.Schema, resource interface{}) (string, error) {
+	// handle "x-kubernetes-int-or-string"
+	var enabled bool
+	if ext, ok := schema.VendorExtensible.Extensions[xKubernetesIntOrString]; ok {
+		enabled, ok = ext.(bool)
+		if !ok {
+			return "", fmt.Errorf("xKubernetesIntOrString extension is not a boolean")
+		}
+	}
+
+	if enabled {
+		switch resource.(type) {
+		case string:
+			return "string", nil
+		case int:
+			return "integer", nil
+		default:
+			return "", nil
```

Comment:
Ditto i think we can error here as well
```suggestion
			return "", fmt.Errorf("found `xKubernetesIntOrString` extension but field value is neither a string nor an integer: %v, resourceType)
```